[Site Admin]: Introduce `SiteHub` Component

retrofox opened the pull request:
~# ❗ Branched off from #99754~

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

This PR introduces the `SiteHub` component, a top-level navigation element for Site Admin that displays the site title, a back navigation button, and a shortcut to the command palette.
This implementation is inspired by the Site Editor Hub in WordPress Core but adapted for Site Admin needs.

### Example Usage  

```tsx
<SiteHub
	isTransparent={ false }
	navigationBackLabel="Back to Dashboard"
	navigationBackLink="/dashboard"
/>
```

### Disclaimer: Command Palette Integration  

The `SiteHub` component does not handle command registration or UI management for the Command Palette.
Command definitions and their behavior are managed through the `@wordpress/commands` store.  

This component only dispatches the `open` action to trigger the Command Palette modal.

Closes #99780

## Proposed Changes

*

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* [Site Admin]: Introduce `SiteHub` Component

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

### Testi with storybbok

1. Update just in case

```sh
yarn install
```

2. Run the Storybook app instance for this package
```sh
yarn workspace @automattic/site-admin run storybook
```

3. Use the `Sidebar` story to test the SiteHub component
  * Confirm it shows the site icon
  * Confirm it shows the site Title and Url
  * Confirm it dispatches the `open` commands action when clicking the magnificent icon.

![site-admin-site-hub](https://github.com/user-attachments/assets/9a5ee25a-e1f2-4f2d-9f9c-2fbd0d72d725)

4. Use the SiteHub stories
5. Confirm the `Focus Button On Mount` story focuses on the Site button once the component is rendered

![admin-site-site-hub](https://github.com/user-attachments/assets/df3c6a91-302c-451c-953a-bd20d861e8ad)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

retrofox committed:
add @wordpress/commands pkg dependency

retrofox committed:
mock root/site to expose site title and url

retrofox committed:
mock core commands store open action

retrofox committed:
introduce first copy of SiteHub

retrofox committed:
copy core layout styles about site hub

retrofox committed:
follow the BEM convention

retrofox committed:
jsdoc SiteHub component

retrofox committed:
add README.md to the SiteHub component

retrofox committed:
fix SiteHub instance issue in the Sidebar story

ciampo commented:
Thank you for the ping!

Given that this PR applies changes on top of https://github.com/Automattic/wp-calypso/pull/99754, I assume PR reviews can wait until all the required code is reviewed and merged — but let me know otherwise.

retrofox commented:
> Thank you for the ping!
> 
> Given that this PR applies changes on top of #99754, I assume PR reviews can wait until all the required code is reviewed and merged — but let me know otherwise.

Sorry, my bad. I applied the proper label. The PR description shows that it has branched off from the previous PR.

ilyasfoo commented:
> The PR description shows that it has branched off from the previous PR.

@retrofox Should the PR also be based off the branch?

retrofox commented:
> > The PR description shows that it has branched off from the previous PR.
> 
> @retrofox Should the PR also be based off the branch?

Yes, but actually, I'm waiting to merge the reference branch. It got in many conflicts, too.  We'll merge it soon.

retrofox committed:
fix linting issue

ciampo commented on packages/site-admin/src/components/site-hub/index.tsx:
```diff
@@ -0,0 +1,129 @@
+interface SiteHubProps {
+	isTransparent: boolean;
+
+	navigationBackLabel: string;
+	navigationBackLink: string; // core picks this prop (dashboardLink) from site-admin store.
```
Given that also the `SidebarNavigationScreen` component [exposes similar props](https://github.com/Automattic/wp-calypso/blob/3065902e81c83bc94fdb84d5965990abe38f6b8e/packages/site-admin/src/components/sidebar-navigation-screen/index.tsx#L30-L31):

- do we need to have these props both in `SidebarNavigationScreen` and in `SiteHub` ?
- in case we do, should we use the name prop names for better consistency? (ie. `exitLabel` and `exitLink` ?)

retrofox replied:
Yes, that makes sense.

I want to review and update this PR since it got a bit old compared with https://github.com/Automattic/wp-calypso/pull/99754. However, the bot doesn't give me the chance :-D 

<img width="537" alt="image" src="https://github.com/user-attachments/assets/461dabec-9999-455d-98f8-d0e6813963c8" />

retrofox replied:
> do we need to have these props both in SidebarNavigationScreen and in SiteHub ?

I suggest keeping the props in both components for now. If I’m not mistaken, in core, these components pull the data from the core data store, which ties the implementation more closely to WordPress.

That’s fine for now, but in my opinion, we should consider creating a dedicated settings store for this implementation. It would allow us to store app/site data like title, home, and other relevant details, making it easier to do things like navigate back to the homepage from the Hub component.

> in case we do, should we use the name prop names for better consistency? (ie. exitLabel and exitLink ?)

definitely

ciampo replied:
Let me see if I understand — you're saying that, in the Gutenberg implementation, both components don't expose these props because they both internally read that information from a core data store?

If we just need to store some "static" data (without the need to modify it at runtime), a data store may be overkill — we should investigate better how we expect consumers to use `SidebarNavigationScreen` and `SiteHub` and see if we effectively need to share data, and how to do that

psealock commented:
When running `yarn install && yarn workspace @automattic/site-admin run build` on this branch I get the following error suggesting we need `@types/wordpress__commands` which doesn't exist. Can you reproduce? 

```
src/components/site-hub/index.tsx:4:40 - error TS7016: Could not find a declaration file for module '@wordpress/commands'. '/Users/psealock/Workspace/calypso/node_modules/@wordpress/commands/build/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/wordpress__commands` if it exists or add a new declaration (.d.ts) file containing `declare module '@wordpress/commands';`
```

ilyasfoo commented on packages/site-admin/.storybook/mock-registry/index.ts:
```diff
@@ -12,15 +13,52 @@ type WPDataRegistry = ReturnType< typeof createRegistry >;
+					action( '[`core/commands`] Open Command Center' )();
+					return { type: 'OPEN' };
+				},
+				close: () => {
```
Nit: Close action is never called in storybook. If we don't need it perhaps remove it?

retrofox replied:
We could, yes. I'd like to improve the storybook story to test this action, though.

ilyasfoo reviewed (commented):
Tests well with a nit comment. I was also able to reproduce the error @psealock  mentioned.

retrofox commented:
> When running `yarn install && yarn workspace @automattic/site-admin run build` on this branch I get the following error suggesting we need `@types/wordpress__commands` which doesn't exist. Can you reproduce?
> 
> ```
> src/components/site-hub/index.tsx:4:40 - error TS7016: Could not find a declaration file for module '@wordpress/commands'. '/Users/psealock/Workspace/calypso/node_modules/@wordpress/commands/build/index.js' implicitly has an 'any' type.
>   Try `npm i --save-dev @types/wordpress__commands` if it exists or add a new declaration (.d.ts) file containing `declare module '@wordpress/commands';`
> ```

The thing is, the commands package doesn’t provide type declarations. I think the straightforward solution is to define these types locally, at least for now.

retrofox committed:
Add type declarations for @wordpress/commands pkg
Create temporary type definitions for commands store and hooks until core package includes them

retrofox commented:
@psealock I've added the type declaration for the commands module here b42e4cc3df740c51df6bece7c513879b84dc2e2c

retrofox committed:
site-admin: update @wordpress/commands pak version

retrofox committed:
Move SiteHub button styles from inline to SCSS

retrofox committed:
Add text domain to i18n strings in site-admin README

retrofox committed:
Add text domain to i18n translations

retrofox committed:
fix border-radious in the site button

retrofox committed:
Update SiteHub class name for better specificity
Replace generic components-button selector with site-admin-site-hub__title-button for clearer styling

retrofox committed:
Refactor view mode toggle styles in site-admin layout
Move toggle button styles into parent selector for better organization and reduced specificity

retrofox committed:
add story for SiteHub component

retrofox committed:
Add focus on mount story with action tracking
Create new story to demonstrate and track automatic focus behavior on SiteHub navigation button

retrofox committed:
Rename SiteHub navigation props for better clarity
Change navigationBackLabel/Link to exitLabel/Link to better reflect their purpose

retrofox committed:
Update CHANGELOG.md with initial release details

retrofox committed:
dedupe yarm file because of @types/debug pkg

retrofox committed:
minor changes in changelog file

retrofox committed:
fix linting errors in md files

ilyasfoo reviewed (approved):
Tests well with one last comment. Pre-approving 🚢 

This may not be relevant to the PR, but perhaps worth to fix. The sidebar title weight seems heavier than in Figma (hdK9XLiOefugy2bxalKBN1-fi-1967_88795):

<img width="773" alt="image" src="https://github.com/user-attachments/assets/6ee6cb80-559b-4f5c-8464-0ac59e977369" />

retrofox commented:
Nice one. Let me take a look at the core implementation.

retrofox commented:
> Tests well with one last comment. Pre-approving 🚢
> 
> This may not be relevant to the PR, but perhaps worth to fix. The sidebar title weight seems heavier than in Figma ([hdK9XLiOefugy2bxalKBN1-fi-1967_88795](https://href.li/?https://www.figma.com/file/hdK9XLiOefugy2bxalKBN1/?node-id=1967-88795)):

Let's follow the core here:

**Storybook**

<img width="630" alt="image" src="https://github.com/user-attachments/assets/136c50a0-8cca-4d6a-9a60-a5f690b7ab01" />

**Site Editor**

<img width="664" alt="image" src="https://github.com/user-attachments/assets/edc3ba99-d50c-4cc9-a11a-757f95b570ed" />

retrofox commented:
thanks, @ilyasfoo, for the review 🙇‍♂️ . going to merge today.

ciampo commented:
> > The sidebar title weight seems heavier than in Figma
> Nice one. Let me take a look at the core implementation.

Not sure what spec should win between Figma and core implementation? I'd probably opt for Figma, but I'd also double-check with the design team.

retrofox commented:
> > > The sidebar title weight seems heavier than in Figma
> > > Nice one. Let me take a look at the core implementation.
> 
> Not sure what spec should win between Figma and core implementation? I'd probably opt for Figma, but I'd also double-check with the design team.

I've checked with the design team and opted for the core design. We can change it is we need it later.

ciampo commented:
> > > > The sidebar title weight seems heavier than in Figma
> > > > Nice one. Let me take a look at the core implementation.
> > 
> > 
> > Not sure what spec should win between Figma and core implementation? I'd probably opt for Figma, but I'd also double-check with the design team.
> 
> I've checked with the design team and opted for the core design. We can change it is we need it later.

Good to know — let's follow-up with the design team so that Figma specs are updated too.
